### ISSUE TITLE
Fail to start two Kestrel servers

### ISSUE THREAD

**Issue by andreymir** [aspnet/IISIntegration#749]
I want to start two Kestrel servers in one binary. One should use IIS Integration and another should not. When I run as console application everything work as expected. I have two servers running on different ports. But when I start via IIS both servers attempt to use the same port and the second server fails to start.

**Comment by jkotalik** [comment]
Do you expect to have one Kestrel server going through IIS and the other server directly listening on a port?

**Comment by Tratcher** [comment]
That's going to be tricky. The IIS support is a lit up based on environment variables. You'd have to suppress them between the first and second servers starting.

https://github.com/aspnet/IISIntegration/blob/346315f5c0f5f6ea064fb4f0fffe930d3717d91c/src/Microsoft.AspNetCore.Server.IISIntegration/WebHostBuilderIISExtensions.cs#L42-L47

**Comment by andreymir** [comment]
> Do you expect to have one Kestrel server going through IIS and the other server directly listening on a port?

Yes, that right. One server suppose to serve a public web site through IIS and another will expose endpoint for based monitoring. It should not be accessible from public and the port should not change between application restarts.

I do not call UseIISIntegration for the second server. It is not clear why second server uses settings provided by IIS.

**Comment by Tratcher** [comment]
Are you calling CreateDefaultBuilder? It includes UseIISIntegration.

**Comment by jkotalik** [comment]
@Tratcher also UseIISIntegration can be called via HostingStartup. Can you change configuration to remove the environment variable before the HostingStartup calls UseIISIntegration?

**Comment by andreymir** [comment]
Thanks for your help! Now it works. I create WebHostBuilder from scratch and override IIS configuration parameters:
```
				var builder = new WebHostBuilder()

					// Для сервера с метриками не нужна интеграция с IIS
					// Переопределяем параметры, которые передает IIS в переменных окружения
					.UseSetting(WebHostDefaults.HostingStartupAssembliesKey, null)
					.UseSetting("PORT", null)
					.UseSetting("TOKEN", null)

					.UseContentRoot(Directory.GetCurrentDirectory())
					.Configure(app => { app.UseMetricServer(); })
					.UseKestrel()
					.ConfigureServices((hostingContext, services) =>
					{
						var metricsPort = hostingContext.Configuration.GetValue<int?>("Metrics:ServerPort") ?? 0;
						services.Configure<KestrelServerOptions>(kestrelConfig =>
						{
							kestrelConfig.Listen(IPAddress.Any, metricsPort);
						});
					});

				var metricsHost = builder.Build();
```

The key is this three lines. Not sure if I need to override them all.
```
					.UseSetting(WebHostDefaults.HostingStartupAssembliesKey, null)
					.UseSetting("PORT", null)
					.UseSetting("TOKEN", null)
```

**Comment by andreymir** [comment]
Still, it is not very obvious that ASPNETCORE_PORT and ASPNETCORE_TOKEN environment variables used even if we don't call UseIISIntegration while configuring WebHost.
